shakenfist/daemons/queues.py

```diff
@@ -103,6 +108,10 @@ def instance_preflight(instance_uuid, network):
                      'schedule', 'retry', None,
                      'insufficient resources: ' + str(e))
 
+    if instance.db_entry.get('placement_attempts') > 3:
+        raise exceptions.AbortInstanceStartException(
+            'Too many start attempts.')
+
     try:
         if instance.db_entry.get('requested_placement'):
             candidates = [instance.db_entry.get('requested_placement')]
```

Comment:
Yes, there is a bug in this logic. The list of candidate should exclude the current node. I will fix.